sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -927,9 +927,14 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product with Tre
       redactMapString(map.asCaseSensitiveMap().asScala, maxFields)
     case map: Map[_, _] =>
       redactMapString(map, maxFields)
-    case t: TableSpec =>
-      t.copy(properties = Utils.redact(t.properties).toMap,
-        options = Utils.redact(t.options).toMap) :: Nil
+    case t: UnresolvedTableSpec =>
+      val newOptionsList = Utils.redact(t.optionsExpressions.map { case (key, value) =>
+        (key, value.sql)
+      }).toMap
+      ResolvedTableSpec(
```

Comment:
Why it becomes a `ResolvedTableSpec` here?

Reply:
I updated this to keep the object as an `UnresolvedTableSpec` here instead.